### DISCUSSION TITLE
USB flashdrive and hub detection not working after pico6502-vscode (Hello World) executed?

### DISCUSSION THREAD

**Discussion by Hyenadae** [picocomputer/community#62] [Q&A] [answered]
So I'm setting up my dev environment for the Pico6502 after seeing it work after assembly with a keyboard and to an actual monitor, but I've got a strange issue with USB terminal/uploading 

When I first loaded up Minicom on my Virtualbox Ubuntu 20.04 VM, the pico Status listed my two USB devices, flash drive and hub fine
(My RIA and VGA are at 0.4 and PWR is to the RIA (right pico) and the VGA has an OTG adapter for USB devices)

The first time I executed the build for Hello World to upload it to the pico6502, it had an error due to minicom conflict / stealing it and no write permissions. I rebooted the VM and all went well on the IDE side

Strangely, when I went to the Minicom term again, and did status, it looked like:
VIA 0.4
`A 0.4 (corrupted text, after multiple status requests)
2 USB

I rebooted the pico6502 board and plugged it into my PC again + reattached to the VM, and re-ran the upload code
After running it and the python script did the upload, no errors were reported by VSCode.
So, I checked minicom again, but now I am seeing 0 USB devices after the successful upload but the text has returned to normal

TLDR,
Is it possible for a bugged upload or using pico6502-vscode (not Pico6502-sdk) to kill all USB detection on the VGA chip?
Additionally, I cannot find any mention of the hello world code, despite ls and listing it on the pico through /dev/ttyacm0. Since I built it, shouldn't it be on the RIA's internal filesystem as a boot rom? I'm getting not found/no directory trying to list anything or anywhere

Do I need to remove it from the case and reflash both the Picos due to some corruption, or is there a library missing that removes all the features? My Picos are fresh installs mind you, I just uploaded the UF2s to their boot-drive and started the above setup

**Comment by Hyenadae** [comment]
Wow, that's awkward. After replugging in the stuff again, and holding down the reset button (for 3 seconds this time) the devices popped up.
So the USB detection code is flakey I suppose? It doesn't exactly explain why the RIA version text was corrupted in minicom initially

**Reply by netzerohero** [reply, marked as answer]
Don't leave minicom constantly up. Be sure to exit minicom before using the python tools for loading from your dev-system and into the RP6502.  Both minicom and the python load script/tools use the same /dev/ttyacm0 resource.

At times - I've found USB detection a bit finicky.  When it doubt, don't just use reset-button - pull-power from RP6502 for a harder-reset. :)
